internal/testutils/balancer.go

```diff
@@ -76,6 +77,21 @@ func (tsc *TestSubConn) UpdateState(state balancer.SubConnState) {
 	}
 }
 
+// UpdateStateAndConnectedAddress saves the connected address to state if the
```

Comment:
I'm not convinced that we need this method on the `TestSubConn` type. It only increases the API surface for something that is only used by one test. Instead I would manually set the connected address from the test under question, and use the existing `UpdateState` method with the SubConnState object that contains the connected address.

Reply:
Done.